core/misc/backdrop.js

```diff
@@ -157,24 +157,78 @@ Backdrop.formatString = function(str, args) {
   // Transform arguments before inserting them.
   for (var key in args) {
     if (args.hasOwnProperty(key)) {
-       switch (key.charAt(0)) {
-         // Escaped only.
-         case '@':
-           args[key] = Backdrop.checkPlain(args[key]);
-        break;
-        // Pass-through.
-         case '!':
-           break;
-         // Escaped and placeholder.
-         case '%':
-         default:
-           args[key] = Backdrop.theme('placeholder', args[key]);
-           break;
-       }
-       str = str.replace(key, args[key]);
+      if (args.hasOwnProperty(key)) {
```

Comment:
This double nests the same `if (args.hasOwnProperty(key)) {` statement twice. We already had that condition in Backdrop (see line 159)